weave/trait/power/power_sources_trait.proto

```diff
@@ -187,5 +187,38 @@ message PowerSourcesTrait {
          *
          */
         repeated uint32                                      active_sources = 4;
+
+        // ----------- EVENTS ----------- //
+        message ActiveSourcesChangedEvent {
+                option (wdl.message_type) = EVENT;
+
+                option (wdl.event) = {
+                    id: 1,
+                    event_importance: EVENT_IMPORTANCE_PRODUCTION_STANDARD,
+                    reserved_tag_min: 1,
+                    reserved_tag_max: 2
+                };
+
+                /**
+                 *  This property represents as an array *all* of the _active_
```

Comment:
I don't think you need to copy-paste the entire comment blob here - you can keep this limited to:

"This field conveys the new set of active sources that were made active at the time this event was emitted. This tracks the active_sources property mentioned above"